decoder/vaapidecoder_h264.cpp

```diff
@@ -1520,15 +1521,89 @@ YamiStatus VaapiDecoderH264::decodeCurrent()
     return status;
 }
 
+uint32_t calcMaxDecFrameBufferingNum(const SharedPtr<SPS> sps)
+{
+
+    if (sps->vui_parameters_present_flag
+        && sps->m_vui.bitstream_restriction_flag) {
+        return sps->m_vui.max_dec_frame_buffering;
+    }
+
+    uint32_t maxDpbMbs;
+
+    /*get MaxDpbMbs as Table A-1*/
+    switch (sps->level_idc) {
+    case 9:
+    case 10:
+        maxDpbMbs = 396;
+        break;
+    case 11:
+        maxDpbMbs = 900;
+        break;
+    case 12:
+    case 13:
+    case 20:
+        maxDpbMbs = 2376;
+        break;
+    case 21:
+        maxDpbMbs = 4752;
+        break;
+    case 22:
+    case 30:
+        maxDpbMbs = 8100;
+        break;
+    case 31:
+        maxDpbMbs = 18000;
+        break;
+    case 32:
+        maxDpbMbs = 20480;
+        break;
+    case 40:
+    case 41:
+        maxDpbMbs = 32768;
+        break;
+    case 42:
+        maxDpbMbs = 34816;
+        break;
+    case 50:
+        maxDpbMbs = 110400;
+        break;
+    case 51:
+    case 52:
+        maxDpbMbs = 184320;
+        break;
+    default:
+        ERROR("undefined level_idc");
+        assert(0);
```

Comment:
level_idc is existed in bitstream, it's not input parameter, user can't change it, which is differnent from the case of https://github.com/01org/libyami/issues/439.